t/core/config_util.t

```diff
@@ -103,6 +108,9 @@ __DATA__
             local item, idx1, idx2 = setup()
             util.cancel_clean_handler(item, idx1)
             util.fire_all_clean_handlers(item)
+
+            local item = setup_to_false()
+            util.fire_all_clean_handlers(item)
```

Comment:
Is it possible to test this bug by deleting a key as you described in the issue? It would be great if we could cover it via an integration test which is more readable.